src/binder/bind/bind_projection_clause.cpp

```diff
@@ -74,10 +74,55 @@ static std::pair<expression_vector, std::vector<std::string>> rewriteProjectionI
     return {newExprs, newAliases};
 }
 
+void validateColumnNamesAreUnique(const std::vector<std::string>& columnNames) {
+    auto existColumnNames = std::unordered_set<std::string>();
+    for (auto& name : columnNames) {
+        if (existColumnNames.contains(name)) {
+            throw BinderException(
+                "Multiple result column with the same name " + name + " are not supported.");
+        }
+        existColumnNames.insert(name);
+    }
+}
+
+std::vector<std::string> getColumnNames(const expression_vector& exprs,
+    const std::vector<std::string>& aliases) {
+    std::vector<std::string> columnNames;
+    for (auto i = 0u; i < exprs.size(); ++i) {
+        if (aliases[i].empty()) {
+            columnNames.push_back(exprs[i]->toString());
+        } else {
+            columnNames.push_back(aliases[i]);
+        }
+    }
+    return columnNames;
+}
+
 BoundWithClause Binder::bindWithClause(const WithClause& withClause) {
     auto projectionBody = withClause.getProjectionBody();
-    auto boundProjectionBody = bindProjectionBody(*projectionBody, true /* isWithClause */);
+    auto [projectionExprs, aliases] = bindProjectionList(*projectionBody);
+    // Check all expressions are aliased
+    for (auto& alias : aliases) {
+        if (alias.empty()) {
+            throw BinderException("Expression in WITH must be aliased (use AS).");
+        }
+    }
+    auto columnNames = getColumnNames(projectionExprs, aliases);
+    validateColumnNamesAreUnique(columnNames);
+    // Rewrite projection list
+    auto originalProjectionExprs = projectionExprs;
+    auto originalAliases = aliases;
+    auto [a, b] = rewriteProjectionInWithClause(projectionExprs, aliases);
+    projectionExprs = a;
+    aliases = b;
```

Comment:
Why not just use new variable names `rewrittenProjectionExprs`, and `rewriteenAliases`?